src/components/People.tsx

```diff
@@ -0,0 +1,73 @@
+import { useEffect, useState } from 'react';
+import { Loader } from './Loader';
+import { getPeople } from '../api';
+import { Person } from '../types';
+import { PersonItem } from './PersonItem';
+
+const preparePeople = (people: Person[]) =>
+  people.map(person => ({
+    ...person,
+    mother: people.find(p => p.name === person.motherName),
+    father: people.find(p => p.name === person.fatherName),
+  }));
+
+export const People = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [hasError, setHasError] = useState(false);
+
+  useEffect(() => {
+    setIsLoading(true);
+    setHasError(false);
+
+    getPeople()
+      .then(peopleFromServer => {
+        const preparedPeople = preparePeople(peopleFromServer);
+
+        setPeople(preparedPeople);
+      })
+      .catch(() => setHasError(true))
+      .finally(() => setIsLoading(false));
+  }, []);
+
+  return (
+    <>
+      <h1 className="title">People Page</h1>
+      <div className="block">
+        <div className="box table-container">
+          {(isLoading && <Loader />) ||
+            (hasError && (
+              <p data-cy="peopleLoadingError" className="has-text-danger">
+                Something went wrong
+              </p>
+            )) ||
+            (!people.length ? (
+              <p data-cy="noPeopleMessage">There are no people on the server</p>
+            ) : (
+              <table
+                data-cy="peopleTable"
+                className="table is-striped is-hoverable is-narrow is-fullwidth"
+              >
+                <thead>
+                  <tr>
+                    <th>Name</th>
+                    <th>Sex</th>
+                    <th>Born</th>
+                    <th>Died</th>
+                    <th>Mother</th>
+                    <th>Father</th>
+                  </tr>
+                </thead>
+
+                <tbody>
+                  {people.map(person => (
+                    <PersonItem key={person.slug} person={person} />
+                  ))}
+                </tbody>
+              </table>
+            ))}
```

Comment:
This past is really hard to read and first glance, could you please do a little refactor, so it's more clear what you're rendering on what condition?